멘토님 코드 리뷰용 PR

kdo1999 opened the pull request:
<!--
  템플릿은 아직 PR 작성이 익숙하지 않으신 분들을 위해서 제공하는 가이드입니다!
  리뷰어 또는 이 PR을 보게 될 다른 사람들이 이 PR을 보는데 참고할 수 있는 내용이 있다면 포함해서 작성해주시면 됩니다.
-->

## 📌 과제 설명 <!-- 어떤 걸 만들었는지 대략적으로 설명해주세요 -->

## 👩‍💻 요구 사항과 구현 내용 <!-- 기능을 Commit 별로 잘개 쪼개고, Commit 별로 설명해주세요 -->

## ✅ 피드백 반영사항  <!-- 지난 코드리뷰에서 고친 사항을 적어주세요. 재PR 시에만 사용해 주세요! (재PR 아닌 경우 삭제) -->

## ✅ PR 포인트 & 궁금한 점 <!-- 리뷰어 분들이 집중적으로 보셨으면 하는 내용을 적어주세요 -->

sobaman committed:
feat: 상품 재고 감소 로직 구현 , 상품이 부족할 때 에러코드 추가

sobaman committed:
feat: 상품 재고 감소 로직 구현 , 상품이 부족할 때 에러코드 추가

vdvhk12 committed:
feat: 회원 탈퇴 기능 구현

- 주문이랑 장바구니 고려하지 않은 상태

vdvhk12 committed:
feat: 회원 탈퇴 기능 구현

- 주문이랑 장바구니 고려하지 않은 상태

kdo1999 committed:
feat: 회원 비밀번호 변경 기능 구현 #54

kdo1999 committed:
refactor: 불필요한 필드 제거 #54

kdo1999 committed:
refactor: login 권한 설정 #54

kdo1999 committed:
test: MemberServiceTest.password_change_success 추가 #54

kdo1999 committed:
test: MemberServiceTest.password_not_match_success 추가 #54

kdo1999 committed:
test: MemberControllerTest.password_change_success 추가 #54

kdo1999 committed:
test: MemberControllerTest.password_change_password_valid_success 추가 #54

kdo1999 committed:
test: MemberControllerTest 수정 #54

kdo1999 committed:
test: MemberControllerTest.password_not_match_success 추가 #54

kdo1999 committed:
refactor: MemberConverter 수정 #54

kdo1999 committed:
refactor: MemberService 수정 #54

wjwj910 committed:
refactor: ResponseEntity 생성 방식을 status 메서드로 변경

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/cart/converter/CartConverter.java:
```diff
@@ -0,0 +1,46 @@
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
```
JPA가 아니죠?
일반적으로 이런 유틸리티 클래스는 `protected` 가 아닌 `private`로 생성자를 정의하는게 좀 더 자연스럽습니다.

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/cart/entity/Cart.java:
```diff
@@ -0,0 +1,33 @@
+    private int quantity;
+
+    @Builder
+    public Cart(Long id, Member member, Product product, int quantity) {
```
Lombok을 사용한다면, 확실하게 대체 가능한 것들은 전부 사용해 주세요.

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/cart/exception/CartException.java:
```diff
@@ -0,0 +1,20 @@
+        this.cartErrorCode = cartErrorCode;
+    }
+
+    public HttpStatus getStatus() {
```
lombok 대체

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/cart/repository/CartRepository.java:
```diff
@@ -0,0 +1,9 @@
+
+public interface CartRepository extends JpaRepository<Cart, Long> {
+
+    boolean existsByProductId_IdAndMemberId_Id(Long productId, Long memberId);
```
Google Java Convention/Naver Java Convention 모두에서, 메서드 이름에 언더스코어를 쓰지 말라고 적혀있을거에요. (일반적으로 테스트가 아니라면 메서드 이름에 언더스코어를 쓰지 않습니다.)

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/cart/service/CartService.java:
```diff
@@ -0,0 +1,53 @@
+
+        // 수량이 0 이하인 경우 exception 발생
+        if (quantity <= 0) {
+            throw new CartException(CartErrorCode.INVALID_QUANTITY);
```
현재 이미 Jakarta Validation 으로 검사 하고 있지 않나요?

vdvhk12 committed:
build: 스웨거 의존성 추가

vdvhk12 committed:
build: 스웨거 의존성 추가

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/common/EmailCertification.java:
```diff
@@ -0,0 +1,42 @@
+	public EmailCertification(String certificationCode, String verifyType) {
+		this.certificationCode = certificationCode;
+		this.verifyType = verifyType;
+		this.sendCount = "1";
```
혹시 이걸 String 으로 처리하는 이유가 있나요?

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/common/EmailCertification.java:
```diff
@@ -0,0 +1,42 @@
+		this.sendCount = "1";
+	}
+
+	public void addResendCount() {
```
미사용 메서드들이 상당히 많네요. 확인 후 제거해 주세요.

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/member/dto/MemberSignupForm.java:
```diff
@@ -0,0 +1,76 @@
+	}
+
+	@Override
+	public String getPassword() {
```
record에 getter가 이미 있죠?

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/member/service/MemberService.java:
```diff
@@ -0,0 +1,122 @@
+			.verifyType(VerifyType.SIGNUP.toString())
+			.build();
+
+		Map convertValue = objectMapper.convertValue(emailCertification, Map.class);
```
처음부터 Map으로 만드면 안되나요? 굳이 모든 것들을 객체로 통제할 필요는 없어보입니다.

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/orders/repository/OrdersRepository.java:
```diff
@@ -0,0 +1,29 @@
+
+public interface OrdersRepository extends JpaRepository<Orders, Long> {
+
+    @Query("select distinct o from Orders o " +
```
multi-line string 사용해 주세요.

```java
@Query("""
	SELECT DISTINCT o
    FROM Orders o
    JOIN FETCH o.member m
    ...
    """)
```

wjwj910 committed:
refactor: 장바구니 조회 API에서 PathVariable 제거 및 인증 정보 활용

wjwj910 committed:
refactor: 장바구니 조회 서비스 로직 간소화(불필요 검증 제거)

wjwj910 committed:
test: PathVariable 제거에 따른 코드 수정

wjwj910 committed:
test: 불필요한 테스트 제거

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/orders/service/OrdersService.java:
```diff
@@ -0,0 +1,88 @@
+
+@RequiredArgsConstructor
+@Service
+@Transactional(readOnly = true)
```
가능하면, `@Transactional` 어노테이션은 위에 두지 말아주세요.
- 아예 DB를 안 찌르는 메서드를 호출할 때도 해당 어노테이션에 의해 영향이 갈 수 있음.
- propagation 까지 도입하게 되면 더욱 로직이 꼬일 수 있음.

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/orders/service/OrdersService.java:
```diff
@@ -0,0 +1,88 @@
+        // map 으로 돌려서 <List>ProductInfoDto 로 변환 후 Response 에 저장
+        List<ProductInfoDto> productInfoDtoList = toProductInfoDtos(orders);
+
+        return OrdersResponse.builder()
```
설명했던 대로, 가능한 한 모든 Service 로직 에서 Builder 사용 코드를 완전히 날려주세요.

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/orders/service/OrdersService.java:
```diff
@@ -0,0 +1,88 @@
+                        .ima_url(po.getProduct().getImgUrl())
+                        .quantity(po.getQuantity())
+                        .build())
+                .collect(Collectors.toList());
```
.toList()

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/product/converter/ProductConverter.java:
```diff
@@ -0,0 +1,39 @@
+                .build();
+    }
+
+    public static ProductResponse from(Product product) {
```
가능하면, 오버라이딩 보단 메서드 이름을 통해 input/output 이 예측되도록 만들어주면 좋을 것 같습니다.

vdvhk12 committed:
feat: 스웨거 접속을 위해 시큐리티 설정 변경, 필터에서 url 패턴 관리하는 메서드 수정

vdvhk12 committed:
feat: 스웨거 접속을 위해 시큐리티 설정 변경, 필터에서 url 패턴 관리하는 메서드 수정

sobaman committed:
Merge branch 'main' into feature/O-03

sobaman committed:
refactor: 원격에 올라간 Delete .DS_Store 삭제

sobaman committed:
refactor: 프론트 쪽에 생기는 .DS_Store gitignore 에 추가

vdvhk12 committed:
feat: 스웨거 설정을 위한 SpringDocConfig 추가

vdvhk12 committed:
feat: 스웨거 설정을 위한 SpringDocConfig 추가

vdvhk12 committed:
feat: MemberController, AuthController 예제로 스웨거 설정 추가

vdvhk12 committed:
feat: MemberController, AuthController 예제로 스웨거 설정 추가

vdvhk12 committed:
refactor: url 리스트 위치 상단으로 이동

vdvhk12 committed:
refactor: url 리스트 위치 상단으로 이동

vdvhk12 committed:
Merge pull request #65 from prgrms-be-devcourse/feature/G-07

Feature/G-07 스웨거 설정 추가

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/member/repository/MemberRepository.java:
```diff
@@ -0,0 +1,36 @@
+	 * @param username
+	 * @return Username이 중복이라면 true, 중복이 아니라면 false
+	 */
+	boolean existsByUsername(String username);
```
(이건 리뷰는 아니고..)
작성하는 메서드에 주석을 달아주는 건 매우 좋습니다. 앞으로 프로젝트 진행할 때도 주석 작성 잘 부탁드릴게요 ㅎㅎ

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/member/service/MemberService.java:
```diff
@@ -0,0 +1,122 @@
+	}
+
+	private String generateCertificationUrl(String to, String certificationCode, VerifyType verifyType) {
+		return verifyUrl + "email=" + to + "&certificationCode="
```
String Template을 쓰는 방법을 고려해 보세요~ (추후 설명)

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/member/service/MemberService.java:
```diff
@@ -0,0 +1,122 @@
+		existsNickname(memberModifyForm.nickname());
+
+		return MemberConverter.from(
+			memberRepository.save(Member.from(MemberConverter.of(memberDto, memberModifyForm))));
```
굳이 하나의 객체를 생성하기 위해 두번의 변환을 수행해야 할까요?

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/orders/entity/Orders.java:
```diff
@@ -0,0 +1,70 @@
+    private Member member;
+
+    @OneToMany(mappedBy = "orders", cascade = CascadeType.ALL)
+    private List<ProductOrders> productOrders = new ArrayList<>();
```
(나중에 고려해 봐야 할 것)
하나의 order에 매핑되는 product_order 의 데이터 수가 5000개 정도 된다고 하면, 어떨까요?

wjwj910 committed:
Merge branch 'main' into feature/C-02

100gammmit committed:
feat: name 필드 unique 속성 추가

100gammmit committed:
feat: repository 중복 상품명 존재 여부 반환 메서드 추가

100gammmit committed:
feat: 중복 상품명 검증 예외 추가

EXISTS_NAME(HttpStatus.BAD_REQUEST, "400-2", "중복된 상품 이름입니다.");

100gammmit committed:
feat: service 상품 추가 시도 시 중복 상품명 존재 여부 검사

중복 상품명일 시 400-2예외 반환

100gammmit committed:
feat: 중복 상품 이름 에러코드 파라미터 순서 오류로 인한 수정

100gammmit committed:
test: 중복 상품 이름 등록 시도 관련 테스트 코드 작성

sobaman committed:
test: 주문 컨트롤러 테스트 구현

주문 단건 조회 성공
주문단건조회 실패
- 존재하지 않는 주문
현재 진행중인 주문조회성공
주문생성실패
- 주소정보누락
- 상품수량부적절
- 상품리스트비어있음
주문생성성공

sobaman committed:
refactor: country 필드 오류 메시지를 국가 -> 도로명 주소로 변경

sobaman committed:
refactor: 상품 재고 감소 로직 중복 수정

sobaman committed:
Merge pull request #68 from prgrms-be-devcourse/feature/O-03

Feature/O-03 주문 등록 기능 구현

wjwj910 committed:
Merge pull request #66 from prgrms-be-devcourse/feature/C-02

Feature/C-02 장바구니 조회 기능 구현

kdo1999 committed:
refactor: 인증 메일 발송 URL 권한 설정 #54

kdo1999 committed:
refactor: 인증메일 폼 생성자 제거 #54

kdo1999 committed:
refactor: 도메인 로직 DTO에서 엔티티로 이동 #54

kdo1999 committed:
feat: 인증 메일 발송시 사용할 폼 추가 #54

kdo1999 committed:
refactor: 서버 오류랑 이메일 인증 관련 예외 코드 추가 #54

kdo1999 committed:
refactor: 이메일 발송 부분 로직 수정 #54

kdo1999 committed:
refactor: MailServiceImpl 로직 수정 #54
- 기존에 다른 서비스에서 처리하던 메일 관련 로직을 MailServiceImpl로 모두 이동

kdo1999 committed:
refactor: 인증 코드 발송 URL 권한 설정 #54

kdo1999 committed:
test: 서비스에서 MemberDto 미사용으로 인한 리팩토링 #54

sobaman committed:
feat: 멤버id 와 주문상태 리스트를 받아 최신 수정시간 순으로 정렬되어 주문목록을 반환하는 쿼리 작성

sobaman committed:
feat: 주문 히스토리 컨트롤러와 서비스 구현

100gammmit committed:
Merge branch 'main' into feature/P-01

kdo1999 committed:
feat: 인증 코드 발송 기능 추가 #54

kdo1999 committed:
feat: 인증 코드 발송 기능 추가 #54

100gammmit committed:
Merge pull request #67 from prgrms-be-devcourse/feature/P-01

Feature/P- 01 상품 이름 unique속성 적용 및 상품 등록 시 검증

sobaman committed:
refactor: 최신 수정시간 순으로 정렬 위해 쿼리에 desc 추가

sobaman committed:
test: 모든 주문 목록 조회 및 수정시간순으로 조회

sobaman committed:
test: 주문 히스토리 조회 서비스, 컨트롤러 테스트

100gammmit committed:
feat: repository 특정 id제외하고 상품 이름 중복 체크하는 메서드 추가

100gammmit committed:
feat: service 상품 수정 시 해당 상품 id제외하고 상품 이름 중복 체크하는 로직 추가

100gammmit committed:
test: 상품 수정 시 해당 상품 id제외하고 상품 이름 중복 체크하는 로직 추가로 인한 관련 테스트 추가

vdvhk12 committed:
test: 회원 탈퇴 테스트 추가

vdvhk12 committed:
feat: 회원 탈퇴 기능 구현

- 탈퇴하려는 회원의 주문 정보, 카트 정보를 전부 삭제함.

vdvhk12 committed:
refactor: 트랜젝션 어노테이션 추가

vdvhk12 committed:
Merge pull request #72 from prgrms-be-devcourse/feature/P-05

Feature/P-05 상품 수정 시 해당 상품 id제외하고 상품 이름 중복 체크하는 로직 추가

vdvhk12 committed:
refactor: 트랜젝션 어노테이션 추가

vdvhk12 committed:
refactor: 멤버 서비스에서 삭제 로직 제거

kdo1999 committed:
refactor: 변수명 수정 #54

kdo1999 committed:
test: AuthServiceTest.send_success 추가 #54

kdo1999 committed:
test: AuthServiceTest.send_verify_type_not_match_fail #54

kdo1999 committed:
refactor: AuthService.send 예외 코드 수정 #54

kdo1999 committed:
test: AuthServiceTest.send_too_many_resend_attempts_fail 추가 #54

kdo1999 committed:
refactor: AuthErrorCode 코드 이름 수정 #54

kdo1999 committed:
test: AuthServiceTest.send_too_many_resend_attempts_fail 수정 #54

kdo1999 committed:
test: AuthControllerTest.send_success #54 추가

kdo1999 committed:
test: AuthControllerTest.send_verify_type_not_match_fail 추가 #54

kdo1999 committed:
refactor: GlobalControllerAdvice 예외 발생 시 로그 수정 #54

kdo1999 committed:
test: AuthControllerTest.send_too_many_resend_attempts_fail 추가 #54

sobaman committed:
Merge pull request #74 from prgrms-be-devcourse/feature/0-04

Feature/0-04 주문 히스토리 조회 api 구현

vdvhk12 committed:
Merge pull request #75 from prgrms-be-devcourse/feature/M-05

Feature/M-05 회원 탈퇴 기능

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/member/dto/MemberDto.java:
```diff
@@ -0,0 +1,55 @@
+		this.modifiedAt = modifiedAt;
+	}
+
+	public MemberDto verify() {
```
verify -> 검증이라는 의미를 갖고 있는데, 이건 순수 객체 변환이네요.

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/member/dto/MemberInfoResponse.java:
```diff
@@ -0,0 +1,9 @@
+import lombok.Builder;
+
+@Builder
+public record MemberInfoResponse (String username, String nickname, Address address){
```
스타일 통일이 필요합니다.

Account-4-Review commented on backend/src/main/java/com/example/backend/domain/orders/dto/OrdersForm.java:
```diff
@@ -0,0 +1,38 @@
+            Long productId, // 상품 ID
+
+            @Min(value = 1, message = "수량은 1 이상이어야 합니다.", groups = MinGroup.class)
+            @Max(value = 99999999, message = "수량은 9,999,999개 이하여야 합니다.", groups = MaxGroup.class)
```
언더스코어 추가해서 숫자 가독성을 높일 수 있습니다.

Account-4-Review replied:
public static 으로 뺄 수도 있습니다.

Account-4-Review reviewed (commented):
일단 진행한 리뷰 남깁니다~

kdo1999 committed:
test: rebase 후 반영 안 된 부분 추가 #54

kdo1999 committed:
refactor: 스웨거 어노테이션 추가 #54

wjwj910 committed:
refactor: build 명시화

wjwj910 committed:
style: 장바구니 추가 메서드명 복수형->단수형 수정

wjwj910 committed:
refactor: 액세스 PROTECTED -> PRIVATE

wjwj910 committed:
refactor: 메서드명에 언더스코어 제거 & vaild로 인한 불필요 검증 로직 제거

wjwj910 committed:
style: 메서드명 언더스코어 제거

wjwj910 committed:
test: 메서드명 변경 및 불필요 검증로직 제거로 인한 test코드 수정

wjwj910 committed:
Merge pull request #78 from prgrms-be-devcourse/feature/C-02

Feature/c 02

wjwj910 committed:
feat: 장바구니에 있는 상품의 수량을 수정하는 메서드(updateCartItemQuantity) 추가

wjwj910 committed:
feat: Cart 엔티티에 수량 업데이트 메서드 추가

wjwj910 committed:
feat: 장바구니 수량 수정 요청용 DTO 추가

wjwj910 committed:
feat: 장바구니 상품 수량 수정 로직(updateCartItemQuantity) 구현

wjwj910 committed:
feat: 장바구니 상품 조회를 위한 메서드(findByProductIdAndMemberId) 추가

wjwj910 committed:
feat: 장바구니 예외 코드 추가

wjwj910 committed:
test: CartControllerTest, 장바구니 상품 수량 수정 메서드 테스트 추가

wjwj910 committed:
test: CartRepositoryTest, 장바구니 항목 조회 및 수량 변경 테스트 추가

wjwj910 committed:
test: CartServiceTest, 장바구니 수량 업데이트 관련 예외 처리 테스트 추가

kdo1999 committed:
test: OrdersRepositoryTest.findReadyOrders 추가 #79

vdvhk12 committed:
refactor: 트랜잭션 어노테이션 메서드 위로 수정 #80

kdo1999 committed:
feat: OrdersRepository.findReadyOrders 추가 #79

vdvhk12 committed:
Merge pull request #81 from prgrms-be-devcourse/feature/M-05

Refactor/M-05 트랜잭션 어노테이션 메서드 위로 수정 #80

wjwj910 committed:
refactor: @Transactional로 인한 불필요한 save 제거

wjwj910 committed:
test: CartServiceTest, 장바구니 수량 업데이트 테스트 개선

wjwj910 committed:
test: CartRepositoryTest, 장바구니 항목 조회 및 수량 변경 테스트 개선

kdo1999 committed:
test: OrdersRepositoryTest.bulkUpdateDeliveryStatus 추가 #79

kdo1999 committed:
feat: OrdersRepository.bulkUpdateDeliveryStatus 추가 #79

kdo1999 committed:
feat: SchedulerService 추가 #79

vdvhk12 committed:
refactor: 회원 정보 수정에서 데이터 변환 수 줄임 #82

wjwj910 committed:
Merge branch 'main' into feature/C-03

kdo1999 committed:
refactor: 예외 내부 에러 코드 필드 final로 변경 #54

kdo1999 committed:
refactor: record에 생성자 제거 #54

kdo1999 committed:
Merge pull request #76 from prgrms-be-devcourse/feature/M-04

Feature/M-04

kdo1999 committed:
refactor: SchedulerService 설정 수정 #79

vdvhk12 committed:
Merge pull request #83 from prgrms-be-devcourse/feature/G-03

Feature/G-03

vdvhk12 committed:
Merge pull request #84 from prgrms-be-devcourse/feature/M-03

Refactor/M-03 회원 정보 수정시 컨버터에서 바로 Member로 변환

wjwj910 committed:
Merge branch 'main' into feature/C-03

100gammmit committed:
feat: Service 상품 삭제 로직 추가

삭제 전 주문된 내역이 있는지 조회하며, 있으면 BAD_REQUEST예외 throw

issue: #60

100gammmit committed:
feat: ProductOrdersRepository 상품 id로 해당 상품 주문 내역 존재 여부 검증 메서드 추가

issue: #60

100gammmit committed:
feat: 주문 내역이 존재하는 상품을 삭제 시도할 때 던질 예외 추가

issue: #60

100gammmit committed:
feat: 상품 삭제 api 핸들러 추가

issue: #60

100gammmit committed:
test: 상품 삭제 테스트 추가

- 각 성공 케이스 및 실패 케이스
- 실패 케이스
 - 해당 상품이 존재하지 않음
 - 해당 상품의 주문내역 존재

issue: #60

kdo1999 committed:
test: OrdersRepository.findUsernameByReady 추가 #85

kdo1999 committed:
feat: OrdersRepository.findUsernameByReady 추가 #85

kdo1999 committed:
feat: 배송 시작 메일 발송 기능 추가 #85

kdo1999 committed:
feat: SchedulerService 이메일 발송 기능 추가 #85

100gammmit committed:
refactor: converter 기본생성자 protected -> private로 설정

- 멘토님 피드백 반영

issue: #60

wjwj910 committed:
refator: validationSequence로 groups 추가

wjwj910 committed:
feat:findByProductIdAndMemberId 쿼리 수정(JOIN FETCH)

kdo1999 committed:
refactor: MailConfig 수정 #85

kdo1999 committed:
Merge pull request #88 from prgrms-be-devcourse/feature/G-04

Feature/G-04

kdo1999 committed:
Merge pull request #89 from prgrms-be-devcourse/feature/P-04

Feature/P-04 상품 삭제 구현

wjwj910 committed:
Merge branch 'main' into feature/C-03

wjwj910 committed:
Merge pull request #86 from prgrms-be-devcourse/feature/C-03

Feature/C-03 장바구니 상품 수량 수정 기능 구현

kdo1999 committed:
refactor: SchedulerService 수정

kdo1999 committed:
Merge pull request #92 from prgrms-be-devcourse/feature/G-04

feature/G-04

vdvhk12 committed:
feat: 로그인 성공시 메인 페이지로 리다이렉트 #93

wjwj910 committed:
feat: CartController, 장바구니 상품 삭제 메서드(deleteCartItem) 메서드 추가

wjwj910 committed:
feat: CartDeleteForm, 장바구니 상품 삭제용 DTO 추가

wjwj910 committed:
feat: 장바구니 상품 삭제 로직(deleteCartItem) 구현

sobaman committed:
feat: 배송 상태에 주문 취소 추가

sobaman committed:
feat: 수량 만큼 상품 수량을 복구하는 기능 추가

sobaman committed:
feat: 주문 취소 기능 구현

- 주문 상태가 이미 취소이면 취소 불가
- 주문 상태가 이미 배송중이면 취소 불가
- 취소 작업 시 취소된 주문 수량만큼 상품 수량 복구

kdo1999 committed:
feat: 응답 값 types로 분리 #90

kdo1999 committed:
feat: 페이징 컴포넌트 추가 #90

kdo1999 committed:
feat: config에 Image 설정 추가 #90

kdo1999 committed:
feat: 상품 목록 불러오는 훅 추가

kdo1999 committed:
refactor: Navbar 패키지 이동 #90

kdo1999 committed:
feat: ProductCard 컴포넌트 추가 #90

kdo1999 committed:
refactor: ProductList 컴포넌트 추가

kdo1999 committed:
refactor: UserProvider 패키지 변경 #90

kdo1999 committed:
refactor: layout.tsx 임포트 변경 #90

kdo1999 committed:
refactor: 상품 목록 컴포넌트로 분리 #90

kdo1999 committed:
refactor: SecurityConfig 상품 목록 권한 설정 #90

vdvhk12 committed:
feat: 이메일 인증이 되지 않은 사용자 로그인시 이메일 재전송 버튼 추가 #93

kdo1999 committed:
refactor: ApiResponse 수정 #90

vdvhk12 committed:
Merge pull request #94 from prgrms-be-devcourse/feature/FE/P-01

Feature/FE/P-01

wjwj910 committed:
test: CartControllerTest, 장바구니 상품 삭제 메서드 테스트 추가

wjwj910 committed:
test: CartServiceTest, 장바구니 상품 삭제 관련 예외 처리 테스트 추가

wjwj910 committed:
Merge branch 'main' into feature/C-04

vdvhk12 committed:
refactor: import 경로 수정

kdo1999 committed:
refactor: ProductResponse 응답 필드명 수정 #90

kdo1999 committed:
test: OrderRepositoryTest 수정

kdo1999 committed:
Merge pull request #96 from prgrms-be-devcourse/feature/FE/P-01

Feature/FE/P-01

wjwj910 committed:
Merge branch 'main' into feature/C-04

vdvhk12 committed:
refactor: isClient 삭제

vdvhk12 committed:
feat: 로딩 상태로 중복 요청막는 로직 추가 #93

vdvhk12 committed:
Merge pull request #97 from prgrms-be-devcourse/feature/FE/A-01

Feature/Fe/A-01 로그인 페이지 구현

sobaman committed:
test: 주문 서비스 테스트 완료

- 주문 정상 취소
- 이미 주문 취소 상태이면 취소 불가
- 배송중일때 주문 취소 불가

sobaman committed:
test: 컨트롤러 테스트 완료

- 취소 시 정상 작동

kdo1999 committed:
refactor: 상품 ID 조회 권한 설정 #98

kdo1999 committed:
refactor: types 수정 #98

kdo1999 committed:
refactor: setProducts 타입 검사 추가 #98

kdo1999 committed:
refactor: ProductCard product 변수명 수정 #98

kdo1999 committed:
feat: useProduct 훅 추가 #98

kdo1999 committed:
refactor: ProductList.tsx product.id 변수명 수정 #98

kdo1999 committed:
feat: 상품 상세 페이지 추가 #98

vdvhk12 committed:
fix: 만료된 액세스 토큰으로 로그아웃 요청시 에러 발생 문제 해결

- 리프레시 토큰으로 로그아웃 로직 진행

vdvhk12 committed:
Merge pull request #102 from prgrms-be-devcourse/feature/A-07

A-07 / 만료된 액세스 토큰으로 로그아웃 요청시 에러 발생 문제 해결

kdo1999 committed:
refactor: SecurityConfig 수정 #98

kdo1999 committed:
feat: FilterUtils 수정 #98

sobaman committed:
Merge pull request #100 from prgrms-be-devcourse/feature/O-05

Feature/O 05 주문 취소 기능 구현

wjwj910 committed:
Merge pull request #99 from prgrms-be-devcourse/feature/C-04

Feature/C-04 장바구니 상품 삭제 기능 구현

kdo1999 committed:
refactor: CartForm 수정 #98

vdvhk12 committed:
refactor: 닉네임 변경 시에만 닉네임 검증하도록 수정

vdvhk12 committed:
refactor: 로그아웃 테스트 수정

vdvhk12 committed:
Merge pull request #104 from prgrms-be-devcourse/feature/M-03

Feature/M-03 회원 수정시 닉네임 변경이 있을 때만 닉네임 검증

kdo1999 committed:
feat: 상품 상세 페이지 및 장바구니 기능 구현 #98

100gammmit committed:
feat : 상품 등록, 수정 시 사용할 폼 구현

issue: #95

100gammmit committed:
Merge pull request #105 from prgrms-be-devcourse/feature/FE/P-02

Feature/FE/P-02

kdo1999 committed:
test: 서비스 로직 수정으로 인한 테스트 제거 #98

kdo1999 committed:
Merge pull request #106 from prgrms-be-devcourse/feature/FE/P-02

feature/FE/P-02

kdo1999 committed:
feat: 이메일 인증 구현 #107

kdo1999 committed:
feat: 회원가입 로직 수정 #107

kdo1999 committed:
refactor: AuthService 인증, 발송 로직 수정 #107

kdo1999 committed:
test: AuthService 수정으로 인한 테스트 수정 #107

wjwj910 committed:
refactor: 조회 로직 반환 data, productId 필드 추가

wjwj910 committed:
Merge pull request #109 from prgrms-be-devcourse/feature/C-02-fix

refactor: 조회 로직 반환 data, productId 필드 추가

kdo1999 committed:
Merge pull request #108 from prgrms-be-devcourse/feature/FE/M-02

Feature/FE/M-02

100gammmit committed:
refactor : ROLE_ADMIN도 상품 단건조회 가능하도록 수정

issue: #95

100gammmit committed:
feat : 관리자 메인페이지 및 이어지는 상품 관리 페이지 구현

- 상품 등록 : 바로 등록 폼으로 이동
- 상품 수정 : 상품 리스트에서 상품 선택 후 수정 폼으로 이동
- 상품 삭제 : 상품 리스트에서 상품 선택하여 삭제

issue: #95

100gammmit committed:
Merge pull request #111 from prgrms-be-devcourse/feature/P-06

Feature/P-06 관리자 페이지 구현

vdvhk12 committed:
feat: 장바구니 페이지 구현 #112

- 조회, 수정, 삭제를 할 수 있음.
- 주문 페이지로 이동할 때 장바구니 점보를 넘길 수 있는지 잘 모르겠어서 일단은 페이지 이동만 해둠.
- 주문 페이지에서 다시 조회하면 되지 않을까 싶음?

vdvhk12 committed:
Merge pull request #113 from prgrms-be-devcourse/feature/FE/C-01

Feature/FE/C-01 장바구니 페이지 구현 #112

kdo1999 committed:
feat: 회원 응답 타입 정의 #110

kdo1999 committed:
feat: userMember 훅 추가 #110

kdo1999 committed:
feat: Profile 컴포넌트 추가 #110

kdo1999 committed:
feat: 내 정보 페이지 구현 #110

kdo1999 committed:
feat: 비밀번호 변경 폼 type 정의 #110

kdo1999 committed:
feat: 비밀번호 변경 기능 구현 #110

kdo1999 committed:
feat: 비밀번호 변경 버튼 추가 #110

kdo1999 committed:
feat: 비밀번호 변경 버튼 추가 #110

kdo1999 committed:
feat: 주문 조회 응답 타입 정의 #110

kdo1999 committed:
feat: 내 정보에 주문 정보 추가 #110

kdo1999 committed:
refactor: ProductInfoDto 변환시 이미지 링크 추가 #110

sobaman committed:
주문 페이지

kdo1999 committed:
refactor: 회원 조회로 유저 이름 설정 #110

kdo1999 committed:
Merge pull request #115 from prgrms-be-devcourse/feature/FE/O-01

Feature/FE/O-01 주문 페이지

kdo1999 committed:
Merge pull request #114 from prgrms-be-devcourse/feature/FE/M-03

Feature/FE/M-03

kdo1999 committed:
refactor: 상품 응답 꺼내는 로직 수정

sobaman committed:
Merge pull request #116 from prgrms-be-devcourse/feature/FE/M-03

feature/FE/M-03

sobaman committed:
feat: 이미지 사용 시 example.com 에러 수정

sobaman committed:
fix: 불필요한 옵셔널 disabled 삭제

sobaman committed:
fix: ApiPaginationResponse, ApiResponse 분리

sobaman committed:
feat: 주문 페이지 구현

kdo1999 committed:
Merge pull request #117 from prgrms-be-devcourse/feature/FE/O-02

Feature/fe/O-02 주문 페이지

kdo1999 committed:
feat: 비밀번호 재설정 기능 추가 #118

kdo1999 committed:
refactor: 장바구니 담기 로직 수정 #118

kdo1999 committed:
refactor: 메일 발송, 인증 로직 수정 #118

sobaman committed:
feat: 주소지 변경 기능 추가

sobaman committed:
feat: 주문 api 호출 후 성공 페이지로 이동

sobaman committed:
Merge pull request #120 from prgrms-be-devcourse/feature/fe/O-03

Feature/fe/O-03 주문 페이지 주소변경 , 결제하기

kdo1999 committed:
feat: UserProvider 닉네임 설정으로 변경

kdo1999 committed:
feat: 회원 응답 값에 nickname, role 추가

kdo1999 committed:
refactor: 관리자 페이지 수정

kdo1999 committed:
refactor: 로그인 페이지 수정

kdo1999 committed:
Merge pull request #119 from prgrms-be-devcourse/feature/FE/M-04

FeatureFE/M-04

kdo1999 committed:
refactor: 주문시 장바구니 삭제 기능 추가

kdo1999 committed:
refactor: 주문 요청시 파라미터 변경

kdo1999 committed:
refactor: 로그아웃시 메인으로 이동 기능 추가

kdo1999 committed:
refactor: CartList li key 값 수정

kdo1999 committed:
refactor: 상품 리스트 제거

kdo1999 committed:
refactor: 비밀번호 변경시 passwordEncoder 암호화 설정

wjwj910 committed:
Update README.md
